source/solvers/vof.cc

```diff
@@ -2062,21 +2168,21 @@ VolumeOfFluid<dim>::apply_peeling_wetting(const unsigned int i_bc,
                     {
                       // Get denser/lighter fluid id
                       if (density_1[q] > density_0[q])
-                        {
-                          phase_denser_fluid_q += 1;
-                        }
+                        phase_denser_fluid_q += 1;
                       else
-                        {
-                          phase_lighter_fluid_q += 1;
-                        }
+                        phase_lighter_fluid_q += 1;
 
-                      // Peeling condition on the pressure gradient
+                      // Condition on the pressure gradient
                       for (unsigned int d = 0; d < dim; d++)
                         {
-                          if (pressure_gradients[q][d] < peeling_grad_p)
-                            {
-                              nb_pressure_grad_meet_peel_condition += 1;
-                            }
+                          double pressure_gradient_q_d =
+                            pressure_gradients[q][d] *
+                            fe_face_values_fd.normal_vector(q)[d];
+
+                          if (pressure_gradient_q_d < -1e-5)
```

Comment:
If you want to use a small negative value, I think using -DBL_MIN is a better choice. If you need this 1e-5 value, I think it's better to keep it as a variable as use the variable elsewhere.